### DISCUSSION TITLE
Question regarding the brand analytics search term report -  GET_BRAND_ANALYTICS_SEARCH_TERMS_REPORT

### DISCUSSION THREAD

**Discussion by amitneeman** [amzn/selling-partner-api-models#3684] [General]
Hi,

Im trying to do a search query research for my amazon store, and wanted to use the SP-API in order to automate my process. I have a list of roughly 2000 search terms which users searched for my products, and for each search query I want to fetch the top 3 competitors (Exactly what the search term report does).

However, when trying to use the API, it doesnt provide any way for querying a specific search query, but instead, only allows to insert a time frame, and the results is 2.3gb of search terms which I had to download. The worse part, is that not only it provides many not relevant search queries, its also not returning all of the search queries available through the brand analytics UI (What I mean is that I have search queries which are not returned in the gigantic report, but if I search for them specifically on seller central I can get their data).

So, someone knows how to run the report through the SP-API with a specific search term? or any other suggestion which will help me achieve my goal?

For convenience, im  adding the documentation for the said report from the SP-API official documentation.

https://github.com/amzn/selling-partner-api-models/blob/main/schemas/reports/sellingPartnerSearchTermsReport.json
https://developer-docs.amazon.com/sp-api/docs/report-type-values-analytics#brand-analytics-reports

Thank you!

**Comment by Javadebi** [comment]
Hello,
I had the same problem and contacted the support team and like always they said it is what it is and API does not provide exact report from seller central

**Comment by michelexport** [comment]
Facing the same here, pretty frustrating to have a report that is pretty useless in this format..at least being able to set paratemers for "deparmentName" would be really helpful

**Comment by Surya9630** [comment]
Hello,
I might be a little late here but rather than using GET_BRAND_ANALYTICS_SEARCH_TERMS_REPORT,
you can use Catalog Items API v2022-04-01
specifically  "Search for items by keywords".

**Reply by michelexport** [reply]
point is that this path does not provide some important metrics compared to  GET_BRAND_ANALYTICS_SEARCH_TERMS_REPORT like conversionshare or clickshare

**Reply by Surya9630** [reply]
That is true.
But I think the way it is supposed to be use is-
GET_BRAND_ANALYTICS_SEARCH_TERMS_REPORT ( Contains data on the **top** clicked ASINs by search keyword and department for a marketplace. )
Since this data cannot be changed, It will remain same.
For example if I acquired Q1 from 2023, the data will remain same regardless of me making request at any point of time.
As it happened in the past. So acquire the data keep it, make it all nice and formatted.
Now since the Catalog Items API does provide the method to get the competitor, use it to the ASINs.
And search those ASINs in the data you have gathered through GET_BRAND_ANALYTICS_SEARCH_TERMS_REPORT  with the help of SQL or python or even excel maybe.
I think that would create a flow.

**Reply by michelexport** [reply]
Thanks for sharing this, it is always very useful to get other point of views :)
Any ideas/ways to get pcs sold or total revenus starting from the path you mentioned?
3rd party tools provided wide different ranges, having 1st party datas would be ideal, thanks!

**Reply by Surya9630** [reply]
You are welcome :),
I am assuming by pcs you meant the total quantity sold for your own products.
It highly depends what your case is-
There are different APIs to gather the sales quantity and total revenue.
You can use Orders API (have an limit to 2 years back window) to see the data row wise, what quantity sold where and at which time.
Then there's Sales API(have an limit to 2 years back window), which gives the direct number of quantity sold at any given time period.
Than there are different Reports API, depending what are you specifically wants to get.
Than maybe if you want to go way beyond 2 years, you could also look into Transactional Report (In amazon central it's in Payment repository) (this is a good way I think, pretty fast and accurate). With API it should be Finance API.

**Reply by michelexport** [reply]
Thanks again :)
We meant quantity sold or revenues us$ for any asin on the marketplace, not only referring to our own company.
We are public devs too with experience with Amazon APIs as well :)

**Reply by Surya9630** [reply]
aah That.
Impossible I would like to say.
But you can work around it a bit-
API will never provide that info.
But you can scrap Detail page of ASIN and look info regarding price, 
![image](https://github.com/amzn/selling-partner-api-models/assets/132869792/f97853ab-a150-4484-ad59-39d0c025907a)
sales like this,
Rank as well ,than you might be able to make an estimated figure.
However, be mindful of Amazon's terms of service and legal considerations regarding web scraping.

Or go for tool such as Helium, Jungle scout which provide same information.

**Comment by moamanuldin** [comment]
@amitneeman Hey man, do you think you could upload this generic dataset. I wanna use it in a project.
